### PR TITLE
Implement contact list.

### PR DESCRIPTION
**Pull request by alecpm**

### CODE REVIEW

File: src/vms/views.py [hunk 1]
```diff
@@ -416,6 +425,15 @@ def workparty_registrations(context, request):
...
         }
 
+        if is_leader:
+            pref_phone = contact['npe01__preferredphone']
+            if pref_phone == 'Home':
+                info['phone'] = contact['homephone']
+            if pref_phone == 'Mobile':
+                info['phone'] = contact['mobilephone']
+            if pref_phone == 'Work':
+                info['phone'] = contact['npe01__workphone']
```
Comment (davisagli) [inline src/vms/views.py]: Use get_best_phone (currently in register.py but should probably move to utils.py) to do this; it is tested and handles the cases where the preferred phone is set to a field that is blank.

File: src/vms/views.py [hunk 2]
```diff
@@ -390,8 +392,15 @@ def workparty_registrations(context, request):
         return reg_data
 
     single_day_registration = wp.data['work_party_type']['allow_single_day_registration']
-    start = dateutil.parser.parse(wp.data['start_date'])
-    end = dateutil.parser.parse(wp.data['end_date'])
+    wp_dict['start_date'] = start = dateutil.parser.parse(wp.data['start_date'])
+    wp_dict['end_date'] = end = dateutil.parser.parse(wp.data['end_date'])
```
Comment (davisagli) [inline src/vms/views.py]: fwiw, the WorkParty context now has `start_date` and `end_date` properties that do this parsing

Comment (alecpm) [inline src/vms/views.py]: Fixed.  Good to know.

File: src/vms/views.py (hunk 1 above)
Comment (alecpm) [inline src/vms/views.py]: Fixed and moved to utils

**Pull request merged**